music21/stream/__init__.py

```diff
@@ -13381,21 +13382,37 @@ def mergeScores(self):
     # -------------------------------------------------------------------------
     def write(self, fmt=None, fp=None, **keywords):
         '''
-        Displays an object in a format provided by the fmt argument or, if not
+        Displays an object in a format provided by the `fmt` argument or, if not
         provided, the format set in the user's Environment.
 
         This method overrides the behavior specified in
         :class:`~music21.base.Music21Object` for all formats besides explicit
         lily.x calls.
 
+        Individual files are written for each score; returns the last file written.
+
+        >>> sc1 = stream.Score()
+        >>> sc2 = stream.Score()
+        >>> o = stream.Opus()
+        >>> o.append([sc1, sc2])
+
+        #_DOCS_SHOW >>> o.write()
+        #_DOCS_SHOW PosixPath('/some/temp/path')
         '''
         if fmt is not None and 'lily' in fmt:
             return Stream.write(self, fmt, fp, **keywords)
         elif common.runningUnderIPython():
             return Stream.write(self, fmt, fp, **keywords)
-        else:
-            for s in self.scores:
-                s.write(fmt=fmt, fp=fp, **keywords)
+        post = []
+        for i, s in enumerate(self.scores):
+            if fp is not None:
```

Comment:
I think what we want to do her is factor the fp check out earlier and something like:

```
if fp is None:
    fp = ...whatever the function to get the fp from format is as a Path...
elif isinstance(fp, str):
    fp = pathlib.Path(fp)

fpParent = fp.parent
fpStem = fp.stem
fpSuffix = '.'.join(fp.suffixes)

numZeros = int(math.log10(len(self.scores))) + 1
formatString = f'0{numZeros}d'

for i, s in enumerate(self.scores):
     scoreName = fpStem + f'{i + 1:{formatString}}' + '.' + fpSuffix
     fpToUse = fpParent / scoreName
     ...
```

Something like that should assure that whatever is passed in as fp all the files get the right extension AND also have sequence numbers.  And if no fp is passed in, all the temporary files still have the same name with extensions.

Reply:
All this might be in a helper function that returns the entire list of post, so that .show() can do the right thing with it.

Reply:
Good stuff here. Not sure I follow what `show` would be able to do with it, since show will just write from scratch and then launch, and doesn't take an fp as an argument.

Reply:
Ah True!